app/models/katello/concerns/pulp_database_unit.rb

```diff
@@ -175,15 +181,40 @@ def copy_repository_associations(source_repo, dest_repo)
                     select #{dest_repo.id} as repository_id, pulp_id, #{columns.join(',')} from #{self.table_name}
                     where repository_id = #{source_repo.id} and pulp_id not in (select pulp_id
                     from #{self.table_name} where repository_id = #{dest_repo.id})"
+          ActiveRecord::Base.connection.execute(insert_query)
 
         end
         ActiveRecord::Base.connection.execute(delete_query)
-        ActiveRecord::Base.connection.execute(insert_query)
       end
 
       def with_pulp_id(unit_pulp_ids)
         where('pulp_id in (?)', unit_pulp_ids)
       end
+
+      def db_columns_sync
+        [unit_id_field, backend_identifier_field, :repository_id, :created_at, :updated_at].compact
+      end
+
+      def db_columns_copy
+        [unit_id_field, backend_identifier_field, :repository_id].compact
+      end
+
+      def db_values_copy(source_repo, dest_repo)
```

Comment:
So it looks like this method is ultimately trying to find the units in the source repo that are missing from the destination repo for copying?

Would a query like this find those missing units?

`missing_units = self.repository_association_class.where(repository: source_repo).where.not(repository: dest_repo)`

Then this method just returns:
```
missing_units.map do |unit|
  [unit.unit_id_field, unit.backend_identifier_field, dest_repo.id].compact
end
```

Reply:
I guess the query would need to consider the unit_id_field as well

Reply:
Made a couple of changes based on this..The query above works great and using it..While testing that realized backend_identifier_field is a method invocation causing sql queries of it's own..So moving both invocations outside loop..